value.go

```diff
@@ -894,7 +897,7 @@ func (vlog *valueLog) getFileRLocked(vp valuePointer) (*logFile, error) {
 
 	// Check for valid offset if we are reading from writable log.
 	maxFid := vlog.maxFid
-	if vp.Fid == maxFid {
+	if vp.Fid == maxFid && !vlog.opt.ReadOnly {
```

Comment:
**Explanation**: In read-only mode, we need not check for the `vlog.writableOffset`.

Reply:
Why is this the case? the writable offset should be greater than the read offset always.

Reply:
Yes, it should be. We would not need this change if we update the writable offset in read-only mode as well by iterating over last vlog file.
I was thinking along the lines that this check is for the vlog file that is currently in use(i.e., they may be written to). As we are not checking the validity of offset for the files other than that with the `maxFid`.